File: leetcode/LC/array/0283_move_zeros.py
"""
take position of last non zero element to be inserted.
and insert non zero elements..

after that fill rest of the part with 0 elements..
"""

from typing import List

class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        
        lastoccurance = 0
        for i in range(len(nums)):
            if nums[i] != 0:
                nums[lastoccurance] = nums[i]
                lastoccurance += 1
        
        for i in range(lastoccurance, len(nums)):
            nums[i] = 0
        return nums

File: leetcode/LC/_c++code/array/0283_move_zeros.cpp
/* 
take position of last non zero element to be inserted.
and insert non zero elements..

after that fill rest of the part with 0 elements..

*/

#include <climits>
#include <iostream>
#include <vector>
#include <algorithm>
#include <string>
#include <unordered_map>
using namespace std;

class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int lastNonZeroFountAt = 0;
        
        for(int i = 0; i < nums.size(); i++){
            if(nums[i] != 0){
                nums[lastNonZeroFountAt++] = nums[i];
            }
        }   
        for(int i = lastNonZeroFountAt; i < nums.size() ; i++){
            nums[i] = 0;
        }
        
    }
};